src/main/java/org/alfresco/repo/solr/SOLRAdminClient.java

```diff
@@ -55,68 +61,42 @@
 import org.quartz.SchedulerException;
 import org.quartz.Trigger;
 import org.quartz.TriggerBuilder;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
 import org.springframework.beans.factory.DisposableBean;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.context.ApplicationEventPublisherAware;
-
 /**
  * Provides an interface to the Solr admin APIs, used by the Alfresco Enterprise JMX layer.
  * Also tracks whether Solr is available, sending Spring events when its availability changes.
  * 
  * @since 4.0
  *
  */
-public class SOLRAdminClient implements ApplicationEventPublisherAware, DisposableBean
+public class SOLRAdminClient extends AbstractSolrAdminHTTPClient
+        implements ApplicationEventPublisherAware, DisposableBean, SolrAdminClient 

```

Comment:
Here's the implementation!  It's probably not a good idea to have two classes that differ only in case: `SOLRAdminClient` and `SolrAdminClient`.  Maybe put `Interface` on the end of the interface?  Usually we'd call the implementation `Impl`, but that probably involves changing more code.

Reply:
Probably I chose the best option having all the other class names into account, but we can change more names if required.